dynamic_programming/0_1_knapsack.cpp

```diff
@@ -1,66 +1,128 @@
-// 0-1 Knapsack problem - Dynamic programming
-//#include <bits/stdc++.h>
+/**
+ * @file
+ * @brief Implementation of [0-1 Knapsack Problem]
+ * (https://en.wikipedia.org/wiki/Knapsack_problem)
+ *
+ * @details
+ * Given weights and values of n items, put these items in a knapsack of
+ * capacity W to get the maximum total value in the knapsack. In other words,
+ * given two integer arrays val[0..n-1] and wt[0..n-1] which represent values
+ * and weights associated with n items respectively. Also given an integer W
+ * which represents knapsack capacity, find out the maximum value subset of
+ * val[] such that sum of the weights of this subset is smaller than or equal to
+ * W. You cannot break an item, either pick the complete item or don’t pick it
+ * (0-1 property)
+ *
+ * ### Algorithm
```

Comment:
this stub lines 16 and 17 can be removed

Reply:
I have added algorithm in the comments now.